free-2.md

```diff
@@ -11,41 +11,135 @@ _What need is met by your idea? Who are you helping? What is that person's
+technically savvy people, but to anyone with an interest in taking a more
+active role in managing their finances.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Intuitive automation for trading stocks.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+I imagine the language as a set of commands that can be used to describe a portfolio and a strategy. The most basic commands might look like:
+```
+BUY <x> SHARES OF <company a>
+SELL <y> SHARES OF <company b>
+```
+
+There should also be some logic associated with these simply buy/sell orders:
+```
+BUY <x> SHARES OF <company c> IF THE PRICE FALLS BELOW <dollar amount>
+SELL <y> SHARES OF <company d> FOR <dollar amount>
+```
+
+There should also be a way to interact with the state of the market:
+```
+IF SPREAD(<company e>) IS MORE THAN <dollar amount> THEN <do something>
+```
+
+Finally, the user may not want to specify companies explicitly, but rather execute some strategy for any company that meets a criterion, like:
+```
+FOR COMPANIES WITH SPREAD <dollar amount> AND VOLUME <shares/day> <do something>
+```
+
+I've written these example commands in a natural language-like pseudo-code to
+reflect the intention to make the language accessible to nontechnical people.
+However, I do not think that natural language is the best way to go (see
+AppleScript). One of the challenges in designing the syntax for this language
+would be in striking the best balance between preciseness and readability.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When a program runs, the strategy is executed as described by the user. This
+will mean placing market and limit orders on the user's behalf, monitoring the
+state of the market and reacting as specified, and searching for companies
+which meet the user's criteria for being viable target stocks.
+
+Besides the obvious sorts of syntax errors, there are a variety of runtime
+errors that we could expect. For example, what should happen if the user's
+strategy recommends buying a stock but the user does not have enough liquid
+funds in their account? Since trading strategies are often designed to work of
+a long period of time, programs written in this language will likely be left to
+run unmonitored. Therefore, reporting these errors could be tricky. One
+possible solution would be to send the user a notification, maybe via email,
+alerting them that a problem has occurred and, if possible, asking them to
+choose from a list of potential solutions. If the runtime cannot determine any
+potential solutions, it could simply let the user know which part of their
+strategy caused the error, and the user could patch up that edge case.
```

Comment:
I think this kind of error handling is good. However, I think it might be hard to implement. I'm not to sure how much work the error handling would entail, but I would at least be wary about it.